pages/DynamicPOC/index.tsx

```diff
@@ -124,51 +124,37 @@ const DynamicPOC: React.FC<{ questions: Question[] }> = ({ questions }) => {
    * Modifies question history and routes form depending on whether answer has been changed
    */
   function _handleQuestionExists() {
-    const nextQuestion = getNextQuestion(
-      questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
-    )
+    const nextQuestion = getNextQuestion(currentQuestion, currentAnswer)
+    const nextAnswer = formValues.formAnswers[currentQuestion.id]
+    // if the exisitng question is a radio, and they're not the same answer
```

Comment:
This is a good thought - we don't need to delete all their previous answers if they change a text question

Reply:
oh ngl it was always like this but it was less explicitly stated, we added comment for clarity